airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -650,15 +651,21 @@ def create_declarative_stream(self, model: DeclarativeStreamModel, config: Confi
             and hasattr(model.incremental_sync, "is_client_side_incremental")
             and model.incremental_sync.is_client_side_incremental
         ):
-            supported_slicers = (DatetimeBasedCursor, GlobalSubstreamCursor, PerPartitionCursor)
+            supported_slicers = (DatetimeBasedCursor, GlobalSubstreamCursor, PerPartitionWithGlobalCursor)
             if combined_slicers and not isinstance(combined_slicers, supported_slicers):
-                raise ValueError("Unsupported Slicer is used. PerPartitionCursor should be used here instead")
+                raise ValueError("Unsupported Slicer is used. PerPartitionWithGlobalCursor should be used here instead")
             client_side_incremental_sync = {
-                "date_time_based_cursor": self._create_component_from_model(model=model.incremental_sync, config=config),
-                "per_partition_cursor": combined_slicers if isinstance(combined_slicers, PerPartitionCursor) else None,
-                "is_global_substream_cursor": isinstance(combined_slicers, GlobalSubstreamCursor),
+                "date_time_based_cursor": self._create_component_from_model(model=model.incremental_sync, config=config)
             }
 
+            client_side_incremental_sync.update(
```

Comment:
why do we need to update the `substream_cursor` key via `update()` instead of in the instantiation of the map like we were previously doing?

Reply:
Fixed